source/simulator/assembly.cc

```diff
@@ -1316,6 +1324,7 @@ namespace aspect
           // also have the number of dofs that correspond just to the element for
           // the system we are currently trying to assemble
           const unsigned int advection_dofs_per_cell = data.local_dof_indices.size();
+          const FiniteElement<dim> &fe = scratch.face_finite_element_values->get_fe();
```

Comment:
Introspection does not have access to the finite element, but `SimulatorAccess` has, so I can use that one (consistently in all loops).